SQLite.CodeFirst/Internal/Builder/CreateTableStatementBuilder.cs

```diff
@@ -19,13 +21,19 @@ public CreateTableStatementBuilder(EntitySet entitySet, AssociationTypeContainer
 
         public CreateTableStatement BuildStatement()
         {
+            var nonAutoincrementKeys = entitySet.ElementType.KeyMembers
+                .Where(km => (km as EdmProperty)?.GetCustomAnnotation<AutoincrementAttribute>() == null);
```

Comment:
I think this logic should be moved into the "PrimaryKeyStatementBuilder" and the "PrimaryKeyStatement" as the autoincrement is a part of the primary key.

Actually then its not necessary to have the "AutoincrementConstraint" class and also not the AddAutoincrementConstraintIfNecessary function as the PrimaryKeyStatementBuilder can decide whether to add the AUTOINCREMENT Keyword or not.

I'll apply these refactorings but I'm not sure about the following.

I was wondering how composite primary keys and autoincrement plays together.

"In sqlite, you only get autoincrement behavior when only one integer column is the primary key. composite keys prevent autoincrement from taking effect."
http://stackoverflow.com/questions/6793205/auto-increment-on-composite-primary-key-sqlite3-python [24.02.2017]

So that means if one specifies a composite primary key its just useless to apply the autoincrement-constraint as it will be ignored anyway.

I would suggest to throw an appropriate exception to signalize the developer that this should not be done.
If no exception is thrown this will result in a ignored autoincrement.

The logic would be like (in the PrimaryKeyStatementBuilder):

bool autoincrement = keyMembers.Any(k => (km as EdmProperty)?.GetCustomAnnotation<AutoincrementAttribute>() != null)
if(autoincrement)
{
  if(keyMembers.Count > 1)
  {
     throw new XyException("Helpful exception message");
  }
   return new PrimaryKeyStatement(keyMembers.Select(km => km.Name), true); // "True" stands for Autoincrement
}
else
{
   return new PrimaryKeyStatement(keyMembers.Select(km => km.Name));
}

And the "PrimaryKeyStatement" is extended to optional append the "AUTOINCREMENT"-keyword.

Please let me know what you think about my suggestions.
Do this still satisfy your needs?